presentation/matrix_view.py

```diff
@@ -1,80 +1,127 @@
-from typing import Dict, List
+import itertools
```

Comment:
(non-critical) Since you need only the `chain` function, I suggest to do `from itertools import chain`